### PR TITLE
Add FhirResourceUpdate sample.

### PR DESCRIPTION
**Pull request by dzlier-gcp**
- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [x] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: healthcare/v1/src/main/java/snippets/healthcare/fhir/resources/FhirResourceUpdate.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    // other types of FHIR resources. If necessary, supply new values for data that
+    // correspond to the FHIR resource you are patching.
+    // String data = "[{\"op\": \"replace\", \"path\": \"/active\", \"value\": false}]";
+
+    // Initialize the client, which will be used to interact with the service.
+    CloudHealthcare client = createClient();
+
+    HttpClient httpClient = HttpClients.createDefault();
+    String uri = String.format("%sv1/%s", client.getRootUrl(), resourceName);
+    URIBuilder uriBuilder = new URIBuilder(uri).setParameter("access_token", getAccessToken());
```
Comment (lesv) [inline healthcare/v1/src/main/java/snippets/healthcare/fhir/resources/FhirResourceUpdate.java:63]: I hate this, but can't think of a decent suggestion on fixing.  Not sure why you hardcode v1 in this instance either, well, I can guess.

File: healthcare/v1/src/main/java/snippets/healthcare/fhir/resources/FhirResourceUpdate.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    StringEntity requestEntity = new StringEntity(data);
+
+    HttpUriRequest request =
+        RequestBuilder.put(uriBuilder.build())
+            .setEntity(requestEntity)
+            .addHeader("Content-Type", "application/fhir+json;charset=utf-8")
+            .addHeader("Accept-Charset", "utf-8")
+            .addHeader("Accept", "application/fhir+json; charset=utf-8")
```
Comment (lesv) [inline healthcare/v1/src/main/java/snippets/healthcare/fhir/resources/FhirResourceUpdate.java:71]: Seems like all of these should be constants in the client library.

**Review by lesv — APPROVED** [review]

**Comment by lesv** [comment]
Well, Java 11 failed, so, please fix that.

File: healthcare/v1/src/main/java/snippets/healthcare/fhir/resources/FhirResourceUpdate.java (hunk 1 above)
Comment (dzlier-gcp) [reply healthcare/v1/src/main/java/snippets/healthcare/fhir/resources/FhirResourceUpdate.java:63]: Copied from FhirResourcePatch lol. I can make improvements though.

**Pull request merged**